packages/app/src/sandbox/index.ts

```diff
@@ -68,6 +72,14 @@ requirePolyfills().then(() => {
             data: {},
           });
         }
+      } else if (data.type === 'sign-in') {
+        await requestSandpackSecretFromApp(data.teamId);
+
+        window.location.reload();
```

Comment:
Once it gets a sandpack-token, it reloads the entire iframe, which will ensure the security headers policy will be applied by the worker